executing/executing.py

```diff
@@ -509,9 +530,556 @@ def visit_ClassDef(self, node):
+            # LOAD_METHOD could load "".join for long "..."%(...) BinOps
+            # this can only be associated by using all positions
+            if self.opname(lasti) in (
+                "LOAD_METHOD",
+                "LOAD_ATTR",
+                "STORE_ATTR",
+                "DELETE_ATTR",
+            ):
+                # lineno and col_offset of LOAD_METHOD and *_ATTR instructions get set to the beginning of
+                # the attribute by the python compiler to improved error messages (PEP-657)
+                # we ignore here the start position and try to find the ast-node just by end position and expected node type
+                # This is save, because there can only be one attribute ending at a specific point in the source code.
+                node = self.find_node(
+                lasti,
+                match_positions=("end_col_offset", "end_lineno"),
+                typ=ast.Attribute,
+            )
+            elif self.opname(lasti) == "CALL":
+                # A CALL instruction can be a method call, in which case the lineno and col_offset gets changed by the compiler.
+                # Therefore we ignoring here this attributes and searchnig for a Call-node only by end_col_offset and end_lineno. 
+                # This is save, because there can only be one method ending at a specific point in the source code.
+                # One closing ) only belongs to one method.
+                node = self.find_node(
+                    lasti,
+                    match_positions=("end_col_offset", "end_lineno"),
+                    typ=ast.Call,
+                )
+            else:
+                raise
+
+        # report KnownIssues
+
+        instruction = self.instruction(lasti)
+
+        if instruction.opname in ("COMPARE_OP", "IS_OP", "CONTAINS_OP") and isinstance(
+            node, self.types_cmp_issue
+        ):
+            if isinstance(node, self.types_cmp_issue_fix):
+                # this is a workaround for https://github.com/python/cpython/issues/95921
+                # we can fix cases with only on comparison inside the test condition
+                #
+                # we can not fix cases like:
+                # if a<b<c and d<e<f: pass
+                # if (a<b<c)!=d!=e: pass
+                # because we don't know which comparison caused the problem
+
+
+                comparisons = [
+                    n
+                    for n in ast.walk(node.test)
+                    if isinstance(n, ast.Compare) and len(n.ops) > 1
+                ]
+
+                assert_(comparisons, "expected at least one comparison")
+
+                if len(comparisons) == 1:
+                    node = comparisons[0]
+                else:
+                    raise KnownIssue(
+                        "multiple chain comparison inside %s can not be fixed" % (node)
+                    )
+
+            else:
+                # Comprehension and generators get not fixed for now.
```

Comment:
Could be made (I even did it at some point, but there is no commit for it) and think it should work.
But I am not so sure that there are no problems.

I started this fix more as a test to understand the issue and got the idea that it could be fixed in some cases.

The reason here is that I don't trust the way I fix the code completely, because it depends on the assumption that I understand the bug correctly. The fix also depends on the behaviour of the bug. 

Normal statements like `if` `while` and so on are straight forward because the broken chained comparison appear only at the conditions.

Comprehensions like `[a for a in l if 5<a[0]<10 if 20<a[1]]` have multiple conditions and I am not so certain about all the complexity that can emerge here. Think also about longer chains with multiple `for`s.

I also want to make clear that there are no cpython bugs in code like this:
```
if a<5:
    pass
```
I saw no issues with "one comparison" only for chained comparisons with more than one operator.

I would much more prefer to fix the cpython issue. I want to rewrite the description because there are not only problems in `assert` statements. I also want to try to fix this issue in cpython, but I don't know how difficult this will be. 

I don't want to make it a requirement for this merge-request.

Summary:
Yes, we could also the Comprehension cases. We can verify our fix with tests, but I think that we would miss the cases which would break our fix, because they might be too wired.
I would much more prefer to spend the time trying to fix the cpython issue.

Reply:
> I would much more prefer to spend the time trying to fix the cpython issue.

While I understand the logic, as a user of executing anxiously waiting to be able to try it with 3.11 which is now in RC2, I would plead with everyone to have a would-be 3.11-compatible version as soon as possible. Right now, I have to skip multiple tests in friendly_traceback when running with 3.11.  If I could run them all my tests, I might find other problems with executing as I have done in the past with some of my 400+ existing unit tests in friendly_traceback.

Reply:
OK, I'm happy to just leave this as is for now.

@aroberge I suggest you clone this fork, checkout the 3.11 branch, and `python-m pip install -e .` in the root folder, where `python` is an interpreter with which you can run friendly-traceback tests.